docs/topics/multiplatform-mobile/multiplatform-mobile-upgrade-app.md

```diff
@@ -0,0 +1,348 @@
+[//]: # (title: Upgrade your app)
+
+<microformat>
+    <p>This is a part of the <strong>Getting started with Kotlin Multiplatform Mobile</strong> tutorial. Before proceeding, make sure you've completed previous steps:</p>
+    <p><img src="icon-1-done.svg" width="20" alt="First step"/> <a href="multiplatform-mobile-setup.md">Set up environment</a><br/><img src="icon-2-done.svg" width="20" alt="Second step"/> <a href="multiplatform-mobile-create-first-app.md">Create your first cross-platform app</a><br/><img src="icon-3-done.svg" width="20" alt="Third step"/> <a href="multiplatform-mobile-dependencies.md">Add dependencies</a><br/><img src="icon-4.svg" width="20" alt="Fourth step"/> <strong>Upgrade your app</strong><br/><img src="icon-5-todo.svg" width="20" alt="Fifth step"/> Wrap up your project</p>
+</microformat>
+
+You've already implemented common logic using external dependencies. Now you can add more complex logic. Network 
+requests and data serialization are the [most popular cases](https://kotlinlang.org/lp/mobile/) to share with Kotlin
+Multiplatform. Learn how to implement them in your first application so that after completing this onboarding journey,
+you can use them in future projects.
+
+The updated app will retrieve data over the internet from a [SpaceX public API](https://docs.spacexdata.com/?version=latest)
+and display the date of the first successful launch of a SpaceX rocket.
+
+## Add more dependencies
+
+You'll need the following multiplatform libraries in your project:
+
+* [`kotlinx.coroutines`](https://github.com/Kotlin/kotlinx.coroutines) for using coroutines to write asynchronous code, thus allowing simultaneous operations
+* [`kotlinx.serialization`](https://github.com/Kotlin/kotlinx.serialization) for deserializing JSON responses into objects of entity classes used to process
+network operations
+* [Ktor](https://ktor.io/) framework as an HTTP client for retrieving data over the internet
+
+### kotlinx.coroutines
+
+To add `kotlinx.coroutines` to your project, specify a dependency in the common source set. For that, add the following
+line to the `build.gradle.kts` file of the shared module:
+
+```kotlin
+sourceSets {
+    commonMain {
+        dependencies {
+            // ...
+           implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core:1.6.2")
+        }
+    }
+}
+```
+
+Multiplatform Gradle plugin automatically adds a dependency to the platform-specific (iOS and Android) parts
+of `kotlinx.coroutines`.
+
+You'll also use the new memory manager for Kotlin/Native, which will become default soon.
+Enable it in the same `build.gradle.kts` file:
+
+```kotlin
+val commonMain by getting {
```

Comment:
And here having 'val commonMain by getting {' is confusing: it leaves an impression that the next part should be on the same level as this property, under 'sourceSets', which is not the case. I'd rather say simply "Add the following at the end of the `build.gradle.kts` file.